app/views/admin/registrations/index.html.haml

```diff
@@ -21,70 +21,96 @@
+            %th{ width: '0' } E-Mail
+            %th{ width: '0' }
+              %abbr{ title: 'Code of Conduct' } CoC
+            %th{ width: '0' } Arrival
+            %th{ width: '0' } Departure
+            %th{ width: '0' } Actions
         %tbody
-          - @registrations.each_with_index do |registration, index|
-            %tr
-              %td
-                = registration.id
-              %td
-                = registration.name.present? ? registration.name : registration.username
-                %br
-                - registration.user.roles.where(resource: @conference).each do |role|
-                  %span.label.label-info
-                    = role.name.titleize
-              %td
-                = registration.email
-              - if @code_of_conduct
-                - if registration.accepted_code_of_conduct
-                  %td.text-success.text-center= fa_icon('check', title: 'accepted')
-                - else
-                  %td.text-center.text-warning
-                    = fa_icon('exclamation-circle',
-                      title: 'Has not accepted Code of Conduct')
-              %td
-                - if registration.arrival
-                  = registration.arrival.strftime('%d %b %H:%M')
-                - else
-                  n/a
-              %td
-                - if registration.departure
-                  = registration.departure.strftime('%d %b %H:%M')
-                - else
-                  n/a
-              -if @conference.questions.any?
-                %td
-                  = link_to 'Questions','#', class: 'btn btn-success question-btn', 'data-id' => index, 'data-name' => registration.name
-              %td{ 'data-order' => registration.attended.to_s }
-                = check_box_tag "#{@conference.short_title}_#{registration.id}", registration.id, registration.attended,
-                  class: 'switch-checkbox',
-                  url: toggle_attendance_admin_conference_registration_path(@conference.short_title, id: registration.id)+"?attended="
-                .btn-group
-                  = link_to 'Edit', edit_admin_conference_registration_path(@conference.short_title, id: registration),
-                  method: :get, class: 'btn btn-primary'
-                  = link_to 'Delete', admin_conference_registration_path(@conference.short_title, registration),
-                  method: :delete, class: 'btn btn-danger', data: { confirm: "Do you really want to delete the Registration for #{registration.name}?" }
-  - @registrations.each_with_index do |registration, index|
-    .questions{class: "question#{index}", style: 'display:none;'}
-      = render partial: 'questions', locals: { registration: registration }
 
-.modal.fade{ id: 'questions', 'role' => 'dialog', 'aria-hidden' => 'true' }
-  .modal-dialog
-    .modal-content
-      .modal-header
-        %h3{id: 'question-modal-header'}
-          Questions for Foobar
-      .modal-body#question-modal-body
+:javascript
```

Comment:
I really think you should put this into a .js file in the asset pipeline...

Reply:
Yeah, I go back and forth on that... and here's what I've come up with:

* If it's used on one page, it belongs in that page.
* If it's used more than once, it should be in the asset pipeline.

Reply:
Yeah the only overhead you would generate people downloading those ~80 lines once :-D But okay